### PR TITLE
[fc][test] Add unit tests for Fast client DispatchingAvroGenericStoreClient

### PR DESCRIPTION
**Pull request by m-nagarajan**
## Summary
1. Add unit tests for `DispatchingAvroGenericStoreClient`
2. Fix some metrics based on the new tests
3. cleaned up metrics verification in integration tests
4. Moved some utils function from ` meta/RequestBasedMetadataTest` to `meta/utils/RequestBasedMetadataTestUtils` to be reused for `DispatchingAvroGenericStoreClientTest`

## How was this PR tested?
GH CI

## Does this PR introduce any user-facing changes?
- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: clients/venice-client/src/test/java/com/linkedin/venice/fastclient/DispatchingAvroGenericStoreClientTest.java [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+      assertEquals(getRequestContext.successRequestKeyCount.get(), 1);
+
+      validateRetryMetrics();
+    } finally {
+      tearDown();
+    }
+  }
+
+  @Test
+  public void testGetWithExceptionFromTransportLayer() throws IOException {
```
Comment (xunyin8) [inline clients/venice-client/src/test/java/com/linkedin/venice/fastclient/DispatchingAvroGenericStoreClientTest.java:212]: Is it possible to also write an unit test to reproduce the issue we saw where rejected requests (sending to a host that's not available) is not reported as unhealthy requests?

File: clients/venice-client/src/test/java/com/linkedin/venice/fastclient/meta/utils/RequestBasedMetadataTestUtils.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import com.linkedin.venice.meta.QueryAction;
+import com.linkedin.venice.metadata.response.MetadataResponseRecord;
+import com.linkedin.venice.metadata.response.VersionProperties;
+import com.linkedin.venice.serializer.SerializerDeserializerFactory;
+import io.tehuti.metrics.MetricsRepository;
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+
+
+public class RequestBasedMetadataTestUtils {
```
Comment (xunyin8) [inline clients/venice-client/src/test/java/com/linkedin/venice/fastclient/meta/utils/RequestBasedMetadataTestUtils.java:29]: Thanks for moving this into its own class to be easily re-used.

**Review by xunyin8 — APPROVED** [review]
Thanks Manoj, do you happen to know the current unit test coverage numbers for FC?

**Comment by m-nagarajan** [comment]
> Thanks Manoj, do you happen to know the current unit test coverage numbers for FC?
Based on jacoco code coverage results, this PR increases the code cov for fast client from 62% to 74%.
image

File: clients/venice-client/src/test/java/com/linkedin/venice/fastclient/DispatchingAvroGenericStoreClientTest.java (hunk 1 above)
Comment (m-nagarajan) [reply clients/venice-client/src/test/java/com/linkedin/venice/fastclient/DispatchingAvroGenericStoreClientTest.java:212]: I made a quick attempt at it and looks like I need to spend some time figuring out whats happening there. Will probably add as a separate test and its respective fix later.

**Review by m-nagarajan — COMMENTED** [review]
Thanks Xun for the review.

**Pull request merged**